### PR TITLE
feat(ns): automatically publish Bootstrapping Signaling records

### PR DESCRIPTION
**Pull request by nils-wisiol**
This PR adds support for [DNSSEC Bootstrapping in the latest version](https://www.ietf.org/archive/id/draft-ietf-dnsop-dnssec-bootstrapping-01.html).

Bootstrapping Records are automatically generated using lua, so online signing is required. The private key is supplied to `ns` via `.env` file and must be the same on all name servers using the same name server name.

Bootstrapping Records can be queried at `_dsboot.`domain-name`._signal.`name-server-name (CDS and CDNSKEY).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ Although most configuration is contained in this repository, some external depen
       - `DESEC_NS_APIKEY`: `ns` API key needed for replication operations
       - `DESEC_NS_CARBONSERVER`: pdns `carbon-server` setting (optional)
       - `DESEC_NS_CARBONOURNAME`: pdns `carbon-ourname` setting (optional)
+      - `DESEC_NS_NAME`: name under which this name server will be reached, e.g. ns1.example.com
```
Comment (peterthomassen) [inline README.md]: Couldn't this be optional, too (just like the signing key)?

File: docker-compose.yml [hunk 2]
```diff
@@ -11,6 +11,10 @@ services:
     - DESEC_NS_CARBONSERVER
     - DESEC_NS_CARBONOURNAME
     - DESEC_NS_APIKEY
+    - DESEC_NS_NAME
+    - DESEC_NS_SIGNALING_DOMAIN_ZONE_KEY_B64
```
Comment (peterthomassen) [inline docker-compose.yml]: Variable name should include "private" so people know it's sensitive

File: ns/signaling_domain_zone.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+SIGNALING_DOMAIN="_signal.$DESEC_NS_NAME"
+
+# Create Signaling Domain zone and sign with provided key
+pdnsutil create-zone "$SIGNALING_DOMAIN" || exit  # quit if zone exists
+pdnsutil replace-rrset "$SIGNALING_DOMAIN" "@" SOA \
+    "${DESEC_NS_SIGNALING_DOMAIN_SOA_MNAME}. ${DESEC_NS_SIGNALING_DOMAIN_SOA_RNAME}. 0 86400 3600 2419200 3600"
+KEY_FILE=$(mktemp)
+echo "$DESEC_NS_SIGNALING_DOMAIN_ZONE_KEY_B64" | base64 -d > "$KEY_FILE"
+pdnsutil import-zone-key "$SIGNALING_DOMAIN" "$KEY_FILE"
+rm "$KEY_FILE"
```
Comment (peterthomassen) [inline ns/signaling_domain_zone.sh]: Best practice:
```sh
pdnsutil import-zone-key "$SIGNALING_DOMAIN" <(base64 -d <<< "$DESEC_NS_SIGNALING_DOMAIN_ZONE_KEY_B64")
```

Avoids secret showing up in process list.

**Review by peterthomassen — CHANGES_REQUESTED** [review]
very cool!

Let's do a load test (concurrently query many signaling records), and if that doesn't break the auth, let's deploy!

File: ns/signaling_domain_zone.sh (hunk 3 above)
Comment (nils-wisiol) [reply ns/signaling_domain_zone.sh]: nice

File: docker-compose.yml (hunk 2 above)
Comment (nils-wisiol) [reply docker-compose.yml]: done

File: README.md (hunk 1 above)
Comment (nils-wisiol) [reply README.md]: it is; README is clarified

**Comment by nils-wisiol** [comment]
Addressed all comments.

**Review by peterthomassen — APPROVED** [review]

**Pull request merged**